qiskit_optimization/algorithms/qrao/quantum_random_access_optimizer.py

```diff
@@ -0,0 +1,312 @@
+            samples: The list of ``SolutionSample`` obtained from the optimization algorithm.
+            encoding: The encoding used for the optimization.
+            relaxed_fval: The optimal function value of the relaxed problem.
+            relaxed_result: The result obtained from the underlying minimum eigensolver.
+            rounding_result: The rounding result.
+        """
+        super().__init__(
+            x=x,
+            fval=fval,
+            variables=variables,
+            status=status,
+            raw_results=None,
+            samples=samples,
+        )
+        self._encoding = encoding
+        self._relaxed_fval = relaxed_fval
+        self._relaxed_result = relaxed_result
+        self._rounding_result = rounding_result
+
+    @property
+    def encoding(self) -> QuantumRandomAccessEncoding:
+        """The encoding used for the optimization."""
+        return self._encoding
+
+    @property
+    def relaxed_fval(self) -> float:
+        """The optimal function value of the relaxed problem."""
+        return self._relaxed_fval
+
+    @property
+    def relaxed_result(
+        self,
+    ) -> MinimumEigensolverResult:
+        """The result obtained from the underlying minimum eigensolver."""
+        return self._relaxed_result
+
+    @property
+    def rounding_result(self) -> RoundingResult:
+        """The rounding result."""
+        return self._rounding_result
+
+
+class QuantumRandomAccessOptimizer(OptimizationAlgorithm):
+    """Quantum Random Access Optimizer class."""
+
+    def __init__(
+        self,
+        min_eigen_solver: MinimumEigensolver,
+        max_vars_per_qubit: int = 3,
+        rounding_scheme: RoundingScheme | None = None,
+        *,
+        penalty: float | None = None,
+    ):
+        """
+        Args:
+            min_eigen_solver: The minimum eigensolver to use for solving the relaxed problem.
+            max_vars_per_qubit: The maximum number of decision variables per qubit.
+            rounding_scheme: The rounding scheme.  If ``None`` is provided,
+                ``SemideterministicRounding()`` will be used.
+            penalty: The penalty factor to use for the ``QuadraticProgramToQubo`` converter.
+
+        Raises:
+            ValueError: If the maximum number of variables per qubit is not 1, 2, or 3.
+            TypeError: If the provided minimum eigensolver does not support auxiliary operators.
+        """
+        if max_vars_per_qubit not in (1, 2, 3):
+            raise ValueError("max_vars_per_qubit must be 1, 2, or 3, but was {max_vars_per_qubit}.")
+        self._max_vars_per_qubit = max_vars_per_qubit
+        self.min_eigen_solver = min_eigen_solver
+        # Use ``QuadraticProgramToQubo`` to convert the problem to a QUBO.
+        if rounding_scheme is None:
+            rounding_scheme = SemideterministicRounding()
+        self._rounding_scheme = rounding_scheme
+        self._converters: QuadraticProgramConverter = QuadraticProgramToQubo(
```

Comment:
Do you need the type hint `QuadraticProgramConverter`? This line explicitly sets `QuadraticProgramToQubo`. Is there any mypy error w/o the type hint?